src/IO/ReadBufferFromEncryptedFile.cpp

```diff
@@ -96,6 +96,7 @@ bool ReadBufferFromEncryptedFile::nextImpl()
     working_buffer.resize(bytes_read);
     encryptor.decrypt(encrypted_buffer.data(), bytes_read, working_buffer.begin());
 
+    offset += bytes_read;
```

Comment:
If I understood correctly the problem without that line was that `getPosition()` returned wrong values after calling next() multiple times. It seems it might be better to make a unit-test [here](https://github.com/ClickHouse/ClickHouse/blob/master/src/IO/tests/gtest_file_encryption.cpp) which will read buffer and seek and check position.